app/Http/Controllers/TeamController.php

```diff
@@ -42,6 +44,38 @@ public function create()
         ]);
     }
 
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return Response
+     */
+    public function joinForm()
+    {
+        return Inertia::render('team/Join');
+    }
+
+    /**
+     * Background Route for joining Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function join(Request $request)
+    {
+        $data = $this->validate($request, [
+            'code' => ['required', 'string', 'exists:teams,join_code'],
+        ]);
+
+        $team = Team::where('join_code', $data['code'])->first();
```

Comment:
Even if we know (because of the request validation) that the team exist, it is better to user `firstOrFail` (which would trigger a "nice" 404 instead of later having a 500 because of taking the id of `null`)

```suggestion
        $team = Team::where('join_code', $data['code'])->firstOrFail();
```